### PR TITLE
alire: init at 1.2.2

### PR DESCRIPTION
**Pull request by atalii**
###### Description of changes

I've added a package for [Alire](https://alire.ada.dev/). This is a first step to getting a pleasant Ada development experience on NixOS.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by atalii** [comment]
Fixes #199825.

**Comment by majiru** [comment]
Result of `nixpkgs-review pr 244743` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>alire</li>
  </ul>
</details>

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ lib, stdenv, fetchFromGitHub, gprbuild, gnat }:
+
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (atalii) [reply pkgs/development/tools/build-managers/alire/default.nix]: Ah, thanks for that! Fixed in 65085626c4f803d3aab13845643e47f76dce7f59.

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./bin $out
+  '';
+
+  meta = with lib; {
+    description = "A catalog of ready-to-use Ada/SPARK libraries plus a command-line tool (alr) to obtain, build, and incorporate the minto your own projects";
+    homepage = "https://alire.ada.dev";
+    license = licenses.gpl3;
```
[the remark this answers is not in the document]
Comment (atalii) [reply pkgs/development/tools/build-managers/alire/default.nix:32]: Of course: 74f1ae7d1f268ad1ff0fc3a6e252b03099121d49.

**Review by majiru — APPROVED** [review]
LGTM!

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./bin $out
+  '';
+
+  meta = with lib; {
+    description = "A catalog of ready-to-use Ada/SPARK libraries plus a command-line tool (alr) to obtain, build, and incorporate them into your own projects";
+    homepage = "https://alire.ada.dev";
+    license = licenses.gpl3;
+    mainteners = with maintainers; [ atalii ];
```
Comment (majiru) [inline pkgs/development/tools/build-managers/alire/default.nix:33]:
```suggestion
    maintainers = with maintainers; [ atalii ];
```
ofborg caught a typo

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r ./bin $out
+  '';
+
+  meta = with lib; {
+    description = "A catalog of ready-to-use Ada/SPARK libraries plus a command-line tool (alr) to obtain, build, and incorporate them into your own projects";
+    homepage = "https://alire.ada.dev";
+    license = licenses.gpl3;
```
Comment (natsukium) [inline pkgs/development/tools/build-managers/alire/default.nix:32]:
```suggestion
    license = licenses.gpl3Only;
```
`licenses.gpl3` is deprecated.
Since it is not clearly stated as `GPL3 or later`, it would be better just to say gpl3Only.

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ gprbuild gnat ];
+
+  # on HEAD (roughly 2c4e5a3), alire provides a dev/build.sh script. for now,
+  # just use gprbuild.
+  buildPhase = ''
+    gprbuild -j0 -P alr_env
```
Comment (natsukium) [inline pkgs/development/tools/build-managers/alire/default.nix:21]:
```suggestion
    gprbuild -j$NIX_BUILD_CORES -P alr_env
```
We can use this variable here.

File: maintainers/maintainer-list.nix
Comment (natsukium) [inline maintainers/maintainer-list.nix:1]: Would you please split this change into a separate commit with the message `maintainers: add atalii`?
Then, please squash other commits with the message `alire: init at 1.2.2`.

**Comment by atalii** [comment]
Thanks for all that, @natsukium - I've applied your feedback and rebased to just the two commits. Please do tell me if there's anything else I'm missing - thanks so much!

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "alire";
+  version = "1.2.2";
+
+  src = fetchFromGitHub {
+    owner = "alire-project";
+    repo = "alire";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-rwNiSXOIIQR1I8wwp1ROVOfEChT6SCa5c6XnTRqekDc=";
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/alire/default.nix:16]:
```suggestion
    hash = "sha256-rwNiSXOIIQR1I8wwp1ROVOfEChT6SCa5c6XnTRqekDc=";
```

File: pkgs/development/tools/build-managers/alire/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A catalog of ready-to-use Ada/SPARK libraries plus a command-line tool (alr) to obtain, build, and incorporate them into your own projects";
+    homepage = "https://alire.ada.dev";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ atalii ];
+  };
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/alire/default.nix:47]:
```suggestion
  meta = {
    description = "A source-based package manager for the Ada and SPARK programming languages";
    homepage = "https://alire.ada.dev";
    changelog = "https://github.com/alire-project/alire/releases/tag/v${finalAttrs.version}";
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [ atalii ];
    platforms = lib.platforms.unix;
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/alire/default.nix (hunk 7 above)
Comment (atalii) [reply pkgs/development/tools/build-managers/alire/default.nix]: Committed; but just to double check: Does lib.platforms.unix imply support for Darwin?

**Comment by AndersonTorres** [comment]
Squash the commits.

**Comment by atalii** [comment]
I think that's all the comments addressed - thanks so much for being patient with me as I figure this all out!
